boards/shared/Test/Src/Test_ErrorTable.cpp

```diff
@@ -86,6 +88,48 @@ class SharedErrorTableTest : public testing::Test
                 error_table, DEFAULT_PDM_NON_CRITICAL_ERROR, true));
     }
 
+    template <typename T>
+    void TestRoutineForSetErrorsFromCanMsg(
+        enum Board                board,
+        std::vector<enum ErrorId> error_ids,
+        uint32_t                  can_stdid,
+        uint32_t                  can_dlc,
+        int (*pack_can_msg)(uint8_t *dst_p, const T *src_p, size_t size),
+        void (*get_boards)(const struct ErrorTable *, struct ErrorBoardList *),
+        void (*get_errors)(struct ErrorTable *, struct ErrorList *))
+    {
+        // Each struct member that has a non-zero value indicates an error that
```

Comment:
Initially, I was assigning each struct member manually but that was incredibly cumbersome and error-prone (For example, we might forget to assign a struct member). Also it didn't allow me to write this function to be shared across different boards.

I think using memset here as a "hack" is justified, especially with the explanation in the comments.